src/bun.js/webcore/streams.zig

```diff
@@ -147,14 +159,19 @@ pub const ReadableStream = struct {
 
     pub fn cancel(this: *const ReadableStream, globalThis: *JSGlobalObject) void {
         JSC.markBinding(@src());
-        this.value.unprotect();
         ReadableStream__cancel(this.value, globalThis);
+        this.value.unprotect();
```

Comment:
Moving this after out of caution that the value gets GC'd immediately after its unprotected